presentation-compiler/src/main/dotty/tools/pc/InferExpectedType.scala

```diff
@@ -0,0 +1,136 @@
+package dotty.tools.pc
+
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.core.StdNames
+import dotty.tools.dotc.core.Symbols
+import dotty.tools.dotc.core.Types.*
+import dotty.tools.dotc.core.Types.Type
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.typer.Applications.UnapplyArgs
+import dotty.tools.dotc.util.NoSourcePosition
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.Spans.Span
+import dotty.tools.pc.IndexedContext
+import dotty.tools.pc.printer.ShortenedTypePrinter
+import dotty.tools.pc.printer.ShortenedTypePrinter.IncludeDefaultParam
+import dotty.tools.pc.utils.InteractiveEnrichments.*
+
+import scala.meta.internal.metals.ReportContext
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.SymbolSearch
+
+class InferExpectedType(
```

Comment:
Should we instead reuse this or parts of it in InferredTypeProvider?

Reply:
It seems that `InferredTypeProvider` still processes `path` to figure out where to add edit and if/what parenthesis should be added. So I'm not sure there is any benefit in this.